source/blog/2015-05-28-rspec-3-3-has-been-released.md

```diff
@@ -0,0 +1,405 @@
+
+Historically, RSpec examples have been identified primarily by _file
+location_. For example, this command:
+
+~~~
+$ rspec spec/unit/baseball_spec.rb:23
+~~~
+
+...would run an example or example group defined on line 23 of
+`spec/unit/baseball_spec.rb`. Location-based identification generally
+works well, but does not always uniquely identity a particular example.
+For example, if you use shared examples, your spec suite may
+have multiple copies of the example defined at
+`spec/support/shared_examples.rb:47`.
+
+RSpec 3.3 introduces a new way to identify examples and example groups:
+unique IDs. The IDs are scoped to a particular file and are based on the
+index of the example or group. For example, this command:
+
+~~~
+$ rspec spec/unit/baseball_spec.rb[1:2,1:4]
+~~~
+
+...would run the 2nd and 4th example or group defined under the 1st
+top-level group defined in `spec/unit/baseball_spec.rb`.
+
+For the most part, the new example IDs are primarily used internally by
+RSpec to support some of the new 3.3 features, but you're free to use
+them from the command line, and the re-run command printed by RSpec for
+each failure will use them if the file location does not uniquely
+identify the failed example. Copying and pasting the re-run command
+for a particular failed example will always run just that example now!
+
+### Core: New `--only-failures` option
+
+Now that RSpec has a robust way to uniquely identify every example, we've
+added new filtering capabilities to allow you to run only the failures. To
+enable this feature, you first have to configure RSpec so it knows where
+to persist the status of each example:
+
+~~~ ruby
+RSpec.configure do |c|
+  c.example_status_persistence_file_path = "./spec/examples.txt"
+end
+~~~
+
+Once you've configured that, RSpec will begin persisting the status of each
+example after every run of your suite. You'll probably want to add this file
+to `.gitignore` (or whatever the equivalent for your source control system is),
+as it's not intended to be kept under source control. With that configuration
+in place, you can use the new CLI option:
+
+~~~
+$ rspec --only-failures
+# or apply it to a specific file or directory:
+$ rspec spec/unit --only-failures
+~~~
+
+It's worth noting that this option filters to the examples that failed
+the last time they ran, not just to the failures from the last run of
+`rspec`. That means, for example, that if there's a slow acceptance spec
+that you don't generally run locally (leaving it to your CI server to
+run it), and it failed the last time it ran locally, even if it was weeks ago,
+it'll be included when you run `rspec --only-failures`.
+
+See our [relish
+docs (TODO)](https://relishapp.com/rspec-staging/rspec-core/docs/command-line/only-failures)
+for an end-to-end example of this feature.
+
+### Core: New `--next-failure` option
+
+When making a change that causes many failures across my spec suite---such
```

Comment:
Not sure what looks odd about it.  It looks right to me.

BTW, I learned of it from working on the book.  @undees, who has authored multiple books for prag prog, informs me it's the correct way to do this in markdown.